src/test/java/io/supertokens/test/userRoles/api/GetUserRolesAPITest.java

```diff
@@ -0,0 +1,158 @@
+            } catch (HttpResponseException e) {
+                System.out.println(e.getMessage());
+                assertTrue(e.statusCode == 400 && e.getMessage().equals(
+                        "Http error. Status Code: 400. Message:" + " Field name 'userId' is missing in GET request"));
+            }
+        }
+
+        {
+            // userId is missing
+            HashMap<String, String> QUERY_PARAM = new HashMap<>();
+            QUERY_PARAM.put("invalid", "invalid");
+            try {
+                HttpRequestForTesting.sendGETRequest(process.getProcess(), "",
+                        "http://localhost:3567/recipe/user/roles", QUERY_PARAM, 1000, 1000, null,
+                        Utils.getCdiVersion2_14ForTests(), "userroles");
+                throw new Exception("should not come here");
+            } catch (HttpResponseException e) {
+                System.out.println(e.getMessage());
+                assertTrue(e.statusCode == 400 && e.getMessage().equals(
+                        "Http error. Status Code: 400. Message:" + " Field name 'userId' is missing in GET request"));
+            }
+        }
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+
+    }
+
+    @Test
+    public void testGettingRolesForAUser() throws Exception {
+        String[] args = { "../" };
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        // create and add multiple roles to a user, get roles for user
+        {
+            String[] roles = new String[] { "role1", "role2", "role3" };
+            String userId = "userId";
+
+            for (String role : roles) {
+                // create role
+                UserRoles.createNewRoleOrModifyItsPermissions(process.main, role, null);
+                // add role to User
+                UserRoles.addRoleToUser(process.main, userId, role);
+            }
+
+            HashMap<String, String> QUERY_PARAMS = new HashMap<>();
+            QUERY_PARAMS.put("userId", userId);
+            JsonObject response = HttpRequestForTesting.sendGETRequest(process.getProcess(), "",
+                    "http://localhost:3567/recipe/user/roles", QUERY_PARAMS, 1000, 1000, null,
+                    Utils.getCdiVersion2_14ForTests(), "userroles");
+
+            assertEquals(2, response.entrySet().size());
+            assertEquals("OK", response.get("status").getAsString());
+
+            JsonArray userRoles = response.getAsJsonArray("roles");
+            assertEquals(roles.length, userRoles.size());
+            userRoles.forEach(role -> {
+                assertTrue(Arrays.asList(roles).contains(role.getAsString()));
+            });
```

Comment:
This is not a good way to test. It will pass even if `userRoles` is {role1, role1, role1}